src/main/java/web/filter/JsonFilter.java

```diff
@@ -0,0 +1,25 @@
+package web.filter;
+
+import javax.servlet.*;
+import javax.servlet.annotation.WebFilter;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@WebFilter(filterName = "JsonFilter", urlPatterns = "/*")
+public class JsonFilter implements Filter {
+    public void destroy() { }
+
+    public void doFilter(ServletRequest req, ServletResponse resp, FilterChain chain) throws ServletException, IOException {
+        if(req.getContentType() == null || req.getContentType().equals("application/json") ){
```

Comment:
Requests `GET` method contain an empty body. So, it does make sense to check their `Content-Type` header